extensions/nns/src/install_nns.rs

```diff
@@ -520,37 +488,24 @@ pub fn upload_nns_sns_wasms_canister_wasms(dfx_cache_path: &Path) -> anyhow::Res
         ..
     } in SNS_CANISTERS
     {
-        let sns_cli = bundled_binary(dfx_cache_path, "sns")?;
-        let wasm_path = nns_wasm_dir(dfx_cache_path).join(wasm_name);
-        let mut command = Command::new(sns_cli);
-        command
-            .arg("add-sns-wasm-for-tests")
-            .arg("--network")
-            .arg("local")
-            .arg("--override-sns-wasm-canister-id-for-tests")
-            .arg(NNS_SNS_WASM.canister_id)
-            .arg("--wasm-file")
-            .arg(&wasm_path)
-            .arg(upload_name);
-        command
-        .stdin(process::Stdio::null())
-        .output()
-            .map_err(anyhow::Error::from)
-            .and_then(|output| {
-                if output.status.success() {
-                    Ok(())
-                } else {
-                    Err(anyhow!(
-                        "Failed to upload {} from {} to the nns-sns-wasm canister:\n{:?} {:?}\nStdout:\n{:?}\n\nStderr:\n{:?}",
-                        upload_name,
-                        wasm_path.to_string_lossy(),
-                        command.get_program(),
-                        command.get_args(),
-                        String::from_utf8_lossy(&output.stdout),
-                        String::from_utf8_lossy(&output.stderr)
-                    ))
-                }
-            })?;
+        let wasm_path = nns_wasm_dir(dfx_cache_path)
+            .join(wasm_name)
+            .display()
+            .to_string();
```

Comment:
Please find a way to do this without a lossy conversion from PathBuf.  `PathBuf.display()` is for display, not for passing to arguments or ultimately using to open a file.

Reply:
turned everything into OsString